src/mbed_tools/build/_internal/cmake_file.py

```diff
@@ -47,7 +47,7 @@ def _render_mbed_config_cmake_template(
     """
     env = jinja2.Environment(loader=jinja2.PackageLoader("mbed_tools.build", str(TEMPLATES_DIRECTORY)),)
     template = env.get_template(TEMPLATE_NAME)
-    options = list(config.options.values())
+    options = [x for x in config.options.values() if not x.key.endswith(".present")]
```

Comment:
Would it not be better to apply this filtering where we create the config "options" here https://github.com/ARMmbed/mbed-tools/blob/495311ca64b90ef068dbc2d9a01e7fb927666615/src/mbed_tools/build/_internal/config/config.py#L106? If not could we at least have a comment as to why we filter this and maybe a `TODO` to remove this when we refactor the mbed config?

Reply:
I'm not sure what refactoring of mbed config you have in mind here. Perhaps we should raise a GitHub issue for it?

Reply:
I meant refactoring of the mbed configuration system to remove these macros from the mbed_lib.json files.